Frameworks/Foundation/NSArray.mm

```diff
@@ -947,21 +923,69 @@ + (NSArray*)arrayWithContentsOfURL:(NSURL*)aURL {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self descriptionWithLocale:locale indent:0];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)level {
-    UNIMPLEMENTED();
-    return StubReturn();
+    thread_local unsigned int indent = 0;
+    NSMutableString* s = [NSMutableString string];
+    NSString* indentStr = (level == 0) ? @"    " : @"\t";
```

Comment:
I'm torn on how this is supposed to work. Shouldn't it call the next one with the next `level`, and always make sure to indent itself to _this_ `level`? The thread local becomes superfluous, and it's initialized with the wrong value anyway.